data-structures/src/data/structure/linkedlist/LinkedList.java

```diff
@@ -0,0 +1,239 @@
+package data.structure.linkedlist;
+import static  java.util.Objects.*;
+
+public class LinkedList<E> {
+    private class Node {
+        E data;
+        Node next;
+        Node previous;
+
+        Node(final E data) {
+            this.data = data;
+            this.next = null;
+            this.previous = null;
+        }
+    }
+
+    private Node head;
+    private Node tail;
+    private int size;
+
+    public LinkedList() {
+        this.head = null;
+        this.tail = null;
+        size = 0;
+    }
+
+    public void add(E element) {

```

Comment:
The `element` parameter could be `final`. Please check the other methods as well.